drivers/bluetooth/hci/h4.c

```diff
@@ -272,6 +272,8 @@ static inline void read_payload(void)
 	}
 
 	prio = (rx.type == H4_EVT && bt_hci_evt_is_prio(rx.evt.evt));
+	normal = (rx.type == H4_ACL ||
```

Comment:
Alternative suggestion, use a recv type and demux to both bt_recv_prio and bt_recv as suitable.

```
enum bt_recv_type {
    BT_RECV_TYPE_NORMAL = 0x00,
    BT_RECV_TYPE_PRIO = 0x02,
    BT_RECV_TYPE_BOTH = 0x03
}

u8_t bt_hci_evt_recv_type(u8_t evt)
{
    switch (evt) {
    case BT_HCI_EVT_CMD_COMPLETE:
    case BT_HCI_EVT_CMD_STATUS:
        /* fallthrough */
#if defined(CONFIG_BT_CONN)
    case BT_HCI_EVT_NUM_COMPLETED_PACKETS:
    case BT_HCI_EVT_DATA_BUF_OVERFLOW:
#endif
        return BT_RECV_TYPE_PRIO;

#if !defined(CONFIG_BT_HCI_RAW)
    case BT_HCI_EVT_DISCONN_COMPLETE:
        return BT_RECV_TYPE_BOTH;
#endif
    }
    return BT_RECV_TYPE_NORMAL;
}
```

in the drivers:
```
if (rx.type == H4_EVT) {
    recv_type = bt_hci_evt_recv_type(rx.evt.evt);
} else {
    recv_type = BT_RECV_TYPE_NORMAL;
}

switch (recv_type) {
case BT_RECV_TYPE_PRIO:
    bt_recv_prio(buf);
    break;
case BT_RECV_TYPE_BOTH:
    bt_recv_prio(buf);
    /* fall through */
case BT_RECV_TYPE_NORMAL:
default:
    bt_recv(buf);
    break;
}
```

Reply:
At some point before the PR was posted I suggested the same, and I think I still prefer that now, so I agree with you @cvinayak. 
I'd like to get @jhedberg's opinion as well.

Reply:
That said I think instead of an enum we could use a mask to simplify checks?

```
#define BT_EVT_TYPE_PRIO BIT(0)
#define BT_EVT_TYPE_NORMAL BIT(1)

if (type & BT_EVT_TYPE_PRIO) bt_recv_prio();
if (type & BT_EVT_TYPE_NORMAL) bt_recv_normal();